### PR TITLE
Feature: DoD webui support

### PR DESCRIPTION
**Pull request by mdeniz**
CRUD operations for Download on Demand repositories in the UI.

### CODE REVIEW

**Comment by ChrisBr** [comment]
Please have a look at the hakiri report
https://hakiri.io/github/openSUSE/open-build-service/master/967051a8fbb3b538402c5cb6f02ab6ddf24c6833/warnings?name=Mass+Assignment

File: src/api/app/controllers/webui/project_controller.rb [hunk 1]
```diff
@@ -869,6 +870,42 @@ def unlock
     end
   end
 
+  def add_dod
+    authorize @project, :add_dod?
+
+    @download_on_demand = DownloadRepository.new(create_dod_params)
+    if @download_on_demand.save
+      @project.store
+      redirect_to project_repositories_path(@project), notice: "Successfully created DoD"
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/project_controller.rb]: How do we name it in the UI? I'm not sure if a user know what ``DoD`` means...

File: src/api/app/controllers/webui/project_controller.rb [hunk 2]
```diff
@@ -888,6 +925,30 @@ def project_params
...
+      :archfilter,
+      :masterurl,
+      :mastersslfingerprint,
+      :pubkey
+    )
+  end
+
+  def update_dod_params
+    params.require(:dod).permit(
+      :repotype,
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/project_controller.rb]: It's not possible to change the arch afterwards, right?

File: src/api/app/policies/project_policy.rb [hunk 3]
```diff
@@ -19,4 +19,16 @@ def destroy?
   def unlock?
     @user.can_modify_project?(@record, true)
   end
+
+  def add_dod?
+    @user.is_admin?
```
Comment (ChrisBr) [inline src/api/app/policies/project_policy.rb]: Atm it's only an admin feature?

Comment (mdeniz) [inline src/api/app/policies/project_policy.rb]: Yes, only admins are allowed

File: src/api/app/controllers/webui/project_controller.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/app/controllers/webui/project_controller.rb]: In theory you may be able, so I will change it.

File: src/api/app/controllers/webui/project_controller.rb (hunk 1 above)
Comment (mdeniz) [inline src/api/app/controllers/webui/project_controller.rb]: Ok, I will change the texts

**Comment by mdeniz** [comment]
@ChrisBr can you please review it?

**Comment by ChrisBr** [comment]
Why did the coverage drop by 20%?

**Comment by mdeniz** [comment]
No idea. Coveralls seems to be not working very well :disappointed:

**Comment by bgeuken** [comment]
THat happened for all PRs, because of master. Most likely caused by c057628ab5d7b24410d502d11c496944ba2764d2

**Comment by mdeniz** [comment]
@bgeuken can you review the PR too?

**Comment by bgeuken** [comment]
Personally I don't like the usage of &nbsp; for styling. I would also avoid inline css and js. But afaik we don't have a styleguide for that. So just my personal taste:-)

**Comment by mdeniz** [comment]
@bgeuken, @hennevogel, @ChrisBr I have changed it to a more "Rails way" shape... Can you review it please?

**Comment by hennevogel** [comment]
LGTM :+1:

**Comment by bgeuken** [comment]
Looks much better now :+1: I still don't like the extensive use of \&nbsp;

**Pull request merged**